minidump/src/minidump.rs

```diff
@@ -2626,6 +2630,51 @@ impl MinidumpSystemInfo {
     pub fn cpu_info(&self) -> Option<Cow<str>> {
         self.cpu_info.as_deref().map(Cow::Borrowed)
     }
+
+    /// Strings identifying the version and build number of the operating
+    /// system.
+    ///
+    /// Tries to parse the version number from the build if it cannot be found
+    /// in the version string. If the stream already contains a valid version
+    /// number or parsing from the build string fails, this will return what's
+    /// directly stored in the stream.
+    pub fn os_parts(&self) -> (String, Option<String>) {
+        let os_version = format!(
+            "{}.{}.{}",
+            self.raw.major_version, self.raw.minor_version, self.raw.build_number
+        );
+
+        let os_build = self
+            .csd_version()
+            .map(|v| v.trim().to_owned())
+            .filter(|v| !v.is_empty());
+
+        if md::PlatformId::from_u32(self.raw.platform_id) != Some(md::PlatformId::Linux)
+            || os_version != "0.0.0"
+        {
+            return (os_version, os_build);
+        }
+
+        // Try to parse the Linux build string. Breakpad and Crashpad run
+        // `uname -srvmo` to generate it. This roughtly resembles:
+        // "Linux [version] [build...] [arch] Linux/GNU"
+        let raw_build = self.csd_version().unwrap_or(Cow::Borrowed(""));
+        let mut parts = raw_build.split(' ');
+        // Skip past "Linux", grab uname -r portion
+        let version = parts.nth(1).unwrap_or("0.0.0");
+        // Start from the end and pull out everything until only the build is left
```

Comment:
this comment threw me off a bit and only after looking at what `uname -v` looks like on several machines did it make sense to me.  I think it's easier to describe this as "trim the operating system (uname -o) and machine (uname -m). of the tail" or something along those lines.